### DISCUSSION TITLE
Impressive!

### DISCUSSION THREAD

**Discussion by KenyOS** [mprimi/portable-secret#2] [General]
I like a lot of the design how you built this, so simple indeed. I was about to study Web Cryptography API can be a good start with your project. I have a few question...
How strong is the cryptography used on the example page?
I guess that if someone use bruteforce method we should garantee a strong password like 12+, right?
it's nice that support files too, but how much I can store inside? kb? mb? gb?

**Comment by mprimi** [comment]
Thank you, Keny!

 > How strong is the cryptography used on the example page?

Encryption is AES-GCM which is a NIST-recommended standard.
Using a password directly as key for AES is not a good idea, entropy would be way too low. 
This is why the password is transformed into a key using PBKDF2.
Still, if someone chose a very simple password (like `banana` in the examples) the output is trivially crackable using dictionary attacks.

The [bounty](https://mprimi.github.io/portable-secret/examples/bounty.html) example is I think closer to what I'd recommend (3-5 words).

But it also depends on your use case. If you want to send an email, and all you care about is Google not reading it, then even using `1234` as password, **and including the password itself in the hint** would still get the job done.

The file size depends on the browser. Specifically the (data URI)[https://en.wikipedia.org/wiki/Data_URI_scheme] generated upon successful decryption. I think for most browsers is a few hundred MBs. But admittedly I have not tried anything bigger than a few MBs.
